tests/acceptance/test_performance_overview.py

```diff
@@ -52,4 +52,5 @@ def test_with_data(self, mock_now):
         with self.feature(FEATURE_NAMES):
             self.browser.get(self.path)
             self.page.wait_until_loaded()
+            self.browser.wait_until('[data-test-id="grid-editable"] td')
```

Comment:
I haven't really looked at this test before but perhaps to reduce flakiness more, you can assert that not only has a grid cell rendered (as might be the case right away with thead?), the last row / cell has finished rendering, and you can do it with wait_until and a little ✨ css. 

iirc, instead of having to find_all the elements and write a check for length, you can wait for the nth element to finish loading by writing a selector to target it only if it's the nth one.  eg. `wait_until('[data-test-id="grid-editable"] tbody tr:last-child:nth-child(50)'`) would wait until the 50th row of the body has loaded.

Reply:
That makes sense - the problem here is more with our event pipeline as the page gets loaded before the event is processed